homeassistant/components/tradfri/__init__.py

```diff
@@ -127,8 +127,19 @@ async def on_hass_stop(event: Event) -> None:
             gateway.get_devices(), timeout=TIMEOUT_API
         )
         devices: list[Device] = await api(devices_commands, timeout=TIMEOUT_API)
-        groups_commands: Command = await api(gateway.get_groups(), timeout=TIMEOUT_API)
-        groups: list[Group] = await api(groups_commands, timeout=TIMEOUT_API)
+
+        if entry.data[CONF_IMPORT_GROUPS]:
+            # Note: we should update this page when deprecating:
+            # https://www.home-assistant.io/integrations/tradfri/
+            _LOGGER.warning(
+                "Importing of Tradfri groups has been deprecated due to stability issues "
+                "and will be removed in Home Assistant core 2022.4"
+            )
+            # No need to load groups if the user hasn't requested it
+            groups_commands: Command = await api(
+                gateway.get_groups(), timeout=TIMEOUT_API
+            )
+            groups: list[Group] = await api(groups_commands, timeout=TIMEOUT_API)
```

Comment:
We still expect this variable to exist below regardless of config option. We'd need a default set somewhere or refactor further.